merlin/models/xgb/__init__.py

```diff
@@ -0,0 +1,133 @@
+from typing import List, Optional
+
+import nvtabular as nvt
+import pandas as pd
+import xgboost as xgb
+
+from merlin.io import Dataset
+from merlin.schema import Tags
+
+
+class XGBoost:
+    """
+    The class adapts an XGBoost model to work with the high level merlin-models API.
+
+    Example usage::
+
+        # get the movielens dataset
+        from merlin.datasets.entertainment import get_movielens
+
+        train, valid = get_movielens()
+
+        # Train an XGBoost model
+        from merlin.schema import Tags
+        from merlin.models.xgb import XGBoost
+
+        model = XGBoost(Tags.REGRESSION)
+        model.fit(train)
+    """
+
+    def __init__(self, target_type: Tags, *args, **kwargs):
```

Comment:
I think it could be more generic if we would allow to user to pass in the objective here. A user could use the `get_objective` utililty to infer this from the target-type.

Reply:
XGBoost objective now in constructor instead of the tag